best-practices/styling/index.html

```diff
@@ -1016,17 +1054,180 @@ <h3>Wrapping and breaking text</h3>
+				<div class="issue" data-number="93" title="Need HTML and CSS of avoiding page breaks between
 				two elements"><span/></div>
-
-				<div class="issue" data-number="93" title="Seeking an example of avoiding orphan or widow
+				<aside class="example" title="THE END kept on last line page so it stays with other material (rather than with a blank before)"></aside>
+				<p>In order to obtain the following desired layout for the above HTML code,</p>
+				<pre><code class="formatted-braille">⠎⠓⠑⠑⠏⠤⠃⠑⠇⠇⠎⠂⠀⠁⠝⠙⠀⠞⠓⠑⠀⠠⠟⠥⠑⠑⠝⠄⠎⠀⠎⠓⠗⠊⠇⠇⠀⠀⠀⠀
+⠉⠗⠊⠑⠎⠀⠞⠕⠀⠞⠓⠑⠀⠧⠕⠊⠉⠑⠀⠕⠋⠀⠞⠓⠑⠀⠎⠓⠑⠏⠓⠑⠗⠙⠀⠀⠀⠀⠀⠀
+⠃⠕⠽⠤⠤⠁⠝⠙⠀⠞⠓⠑⠀⠎⠝⠑⠑⠵⠑⠀⠕⠋⠀⠞⠓⠑⠀⠃⠁⠃⠽⠂⠀⠞⠓⠑⠀⠀⠀⠀
+⠎⠓⠗⠊⠑⠅⠀⠕⠋⠀⠞⠓⠑⠀⠠⠛⠗⠽⠏⠓⠕⠝⠂⠀⠁⠝⠙⠀⠁⠇⠇⠀⠞⠓⠑⠀⠀⠀⠀⠀
+⠕⠞⠓⠑⠗⠀⠟⠥⠑⠑⠗⠀⠝⠕⠊⠎⠑⠎⠂⠀⠺⠕⠥⠇⠙⠀⠉⠓⠁⠝⠛⠑⠀⠐⠣⠎⠓⠑⠀⠀
+⠅⠝⠑⠺⠐⠜⠀⠞⠕⠀⠞⠓⠑⠀⠉⠕⠝⠋⠥⠎⠑⠙⠀⠉⠇⠁⠍⠕⠥⠗⠀⠕⠋⠀⠞⠓⠑⠀⠀⠀
+⠃⠥⠎⠽⠀⠋⠁⠗⠍⠤⠽⠁⠗⠙⠤⠤⠺⠓⠊⠇⠑⠀⠞⠓⠑⠀⠇⠕⠺⠊⠝⠛⠀⠕⠋⠀⠞⠓⠑⠀
+⠉⠁⠞⠞⠇⠑⠀⠊⠝⠀⠞⠓⠑⠀⠙⠊⠎⠞⠁⠝⠉⠑⠀⠺⠕⠥⠇⠙⠀⠞⠁⠅⠑⠀⠞⠓⠑⠀⠀⠀
+⠏⠇⠁⠉⠑⠀⠕⠋⠀⠞⠓⠑⠀⠠⠍⠕⠉⠅⠀⠠⠞⠥⠗⠞⠇⠑⠄⠎⠀⠓⠑⠁⠧⠽⠀⠎⠕⠃⠎⠲
+⠀⠀⠠⠇⠁⠎⠞⠇⠽⠂⠀⠎⠓⠑⠀⠏⠊⠉⠞⠥⠗⠑⠙⠀⠞⠕⠀⠓⠑⠗⠎⠑⠇⠋⠀⠓⠕⠺⠀⠀
+⠞⠓⠊⠎⠀⠎⠁⠍⠑⠀⠇⠊⠞⠞⠇⠑⠀⠎⠊⠎⠞⠑⠗⠀⠕⠋⠀⠓⠑⠗⠎⠀⠺⠕⠥⠇⠙⠂⠀⠀
+⠊⠝⠀⠞⠓⠑⠀⠁⠋⠞⠑⠗⠤⠞⠊⠍⠑⠂⠀⠃⠑⠀⠓⠑⠗⠎⠑⠇⠋⠀⠁⠀⠛⠗⠕⠺⠝⠀⠀⠀
+⠺⠕⠍⠁⠝⠆⠀⠁⠝⠙⠀⠓⠕⠺⠀⠎⠓⠑⠀⠺⠕⠥⠇⠙⠀⠅⠑⠑⠏⠂⠀⠞⠓⠗⠕⠥⠛⠓⠀⠀
+⠁⠇⠇⠀⠓⠑⠗⠀⠗⠊⠏⠑⠗⠀⠽⠑⠁⠗⠎⠂⠀⠞⠓⠑⠀⠎⠊⠍⠏⠇⠑⠀⠁⠝⠙⠀⠀⠀⠀⠀
+⠇⠕⠧⠊⠝⠛⠀⠓⠑⠁⠗⠞⠀⠕⠋⠀⠓⠑⠗⠀⠉⠓⠊⠇⠙⠓⠕⠕⠙⠒⠀⠁⠝⠙⠀⠓⠕⠺⠀⠀	
+⠎⠓⠑⠀⠺⠕⠥⠇⠙⠀⠛⠁⠞⠓⠑⠗⠀⠁⠃⠕⠥⠞⠀⠓⠑⠗⠀⠕⠞⠓⠑⠗⠀⠇⠊⠞⠞⠇⠑⠀
+⠉⠓⠊⠇⠙⠗⠑⠝⠂⠀⠁⠝⠙⠀⠍⠁⠅⠑⠀⠠⠠⠞⠓⠑⠊⠗⠀⠑⠽⠑⠎⠀⠃⠗⠊⠛⠓⠞⠀⠀
+⠁⠝⠙⠀⠑⠁⠛⠑⠗⠀⠺⠊⠞⠓⠀⠍⠁⠝⠽⠀⠁⠀⠎⠞⠗⠁⠝⠛⠑⠀⠞⠁⠇⠑⠂⠀⠀⠀⠀⠀
+⠏⠑⠗⠓⠁⠏⠎⠀⠑⠧⠑⠝⠀⠺⠊⠞⠓⠀⠞⠓⠑⠀⠙⠗⠑⠁⠍⠀⠕⠋⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠠⠺⠕⠝⠙⠑⠗⠇⠁⠝⠙⠀⠕⠋⠀⠇⠕⠝⠛⠀⠁⠛⠕⠒⠀⠁⠝⠙⠀⠓⠕⠺⠀⠎⠓⠑⠀⠀⠀⠀
+⠺⠕⠥⠇⠙⠀⠋⠑⠑⠇⠀⠺⠊⠞⠓⠀⠁⠇⠇⠀⠞⠓⠑⠊⠗⠀⠎⠊⠍⠏⠇⠑⠀⠀⠀⠀⠀⠀⠀⠀
+⠎⠕⠗⠗⠕⠺⠎⠂⠀⠁⠝⠙⠀⠋⠊⠝⠙⠀⠁⠀⠏⠇⠑⠁⠎⠥⠗⠑⠀⠊⠝⠀⠁⠇⠇⠀⠀⠀⠀⠀
+⠞⠓⠑⠊⠗⠀⠎⠊⠍⠏⠇⠑⠀⠚⠕⠽⠎⠂⠀⠗⠑⠍⠑⠍⠃⠑⠗⠊⠝⠛⠀⠓⠑⠗⠀⠕⠺⠝⠀⠀
+⠉⠓⠊⠇⠙⠤⠇⠊⠋⠑⠂⠀⠁⠝⠙⠀⠞⠓⠑⠀⠓⠁⠏⠏⠽⠀⠎⠥⠍⠍⠑⠗⠀⠙⠁⠽⠎⠲⠀⠀
+⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠨⠣⠠⠞⠓⠑⠀⠠⠑⠝⠙⠈⠨⠜⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀</code></pre>
+				</aside>
+				<div class="issue" data-number="93" title="Need HTML and CSS of avoiding orphan or widow
 				lines"><span/></div>
-
+				<aside class="example" title="Heading 2 moving to next page to remain with material"></aside>
+					<p>In order to obtain the following desired layout for the above HTML code,</p>
+					<pre><code class="formatted-braille">⠠⠔⠞⠗⠕⠙⠥⠉⠰⠝⠀⠞⠕⠀⠠⠡⠑⠑⠎⠑
+⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠠⠡⠑⠑⠎⠑⠀⠊⠎⠀⠐⠕⠀⠷⠀⠮⠀⠕⠇⠙⠑⠌⠀⠯⠀⠍⠕⠌⠀⠧⠻⠎⠁⠞⠊⠇⠑⠀
+⠋⠕⠕⠙⠎⠀⠔⠀⠮⠀⠸⠺⠂⠀⠾⠀⠁⠀⠓⠊⠌⠕⠗⠽⠀⠙⠁⠞⠬⠀⠃⠁⠉⠅⠀⠹⠳⠎⠯⠎
+⠷⠀⠽⠑⠜⠎⠲⠀⠠⠍⠁⠙⠑⠀⠏⠗⠊⠍⠜⠊⠇⠽⠀⠋⠀⠍⠊⠇⠅⠂⠀⠡⠑⠑⠎⠑⠀⠉⠀⠀
+⠧⠜⠽⠀⠎⠊⠛⠝⠊⠋⠊⠉⠁⠝⠞⠇⠽⠀⠔⠀⠋⠇⠁⠧⠕⠗⠂⠀⠞⠑⠭⠞⠥⠗⠑⠂⠀⠯⠀⠀
+⠿⠍⠀⠙⠑⠏⠢⠙⠬⠀⠕⠝⠀⠋⠁⠉⠞⠕⠗⠎⠀⠇⠀⠮⠀⠞⠽⠏⠑⠀⠷⠀⠍⠊⠇⠅⠂⠀⠀⠀
+⠁⠛⠬⠀⠏⠗⠕⠉⠑⠎⠎⠂⠀⠯⠀⠏⠗⠕⠙⠥⠉⠰⠝⠀⠞⠑⠡⠝⠊⠟⠥⠑⠎⠲⠀⠀⠀⠀⠀⠀
+⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠀⠀⠠⠞⠽⠏⠑⠎⠀⠷⠀⠠⠍⠊⠇⠅⠀⠠⠥⠎⠫⠀⠔⠀⠠⠡⠑⠑⠎⠑⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠀⠀⠠⠏⠗⠕⠙⠥⠉⠰⠝⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠠⠡⠑⠑⠎⠑⠀⠉⠀⠆⠀⠍⠁⠙⠑⠀⠋⠀⠮⠀⠍⠊⠇⠅⠀⠷⠀⠧⠜⠊⠳⠎⠀⠀⠀⠀⠀⠀⠀⠀
+⠁⠝⠊⠍⠁⠇⠎⠂⠀⠑⠁⠡⠀⠒⠞⠗⠊⠃⠥⠞⠬⠀⠥⠝⠊⠟⠥⠑⠀⠐⠡⠊⠌⠊⠉⠎⠀⠞⠕⠀
+⠮⠀⠋⠔⠁⠇⠀⠏⠗⠕⠙⠥⠉⠞⠲⠀⠠⠮⠀⠍⠕⠌⠀⠉⠕⠍⠍⠕⠝⠀⠞⠽⠏⠑⠎⠀⠷⠀⠀⠀
+⠍⠊⠇⠅⠀⠥⠎⠫⠀⠜⠑⠒⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠠⠉⠪⠄⠎⠀⠠⠍⠊⠇⠅⠒⠀⠠⠮⠀⠍⠕⠌⠀⠺⠊⠙⠑⠇⠽⠀⠥⠎⠫⠀⠍⠊⠇⠅⠀⠀⠀⠀⠀
+⠠⠛⠕⠁⠞⠄⠎⠀⠠⠍⠊⠇⠅⠒⠀⠠⠐⠅⠝⠀⠿⠀⠭⠎⠀⠲⠞⠔⠉⠞⠊⠧⠑⠀⠞⠁⠝⠛⠽⠀
+⠀⠀⠋⠇⠁⠧⠕⠗⠂⠀⠷⠞⠢⠀⠋⠨⠙⠀⠔⠀⠎⠷⠞⠻⠀⠡⠑⠑⠎⠑⠎⠲⠀⠀⠀⠀⠀⠀⠀⠀
+⠠⠩⠑⠑⠏⠄⠎⠀⠠⠍⠊⠇⠅⠒⠀⠠⠗⠊⠡⠀⠔⠀⠋⠁⠞⠂⠀⠹⠀⠍⠊⠇⠅⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠏⠗⠕⠙⠥⠉⠑⠎⠀⠗⠕⠃⠥⠌⠂⠀⠉⠗⠂⠍⠽⠀⠡⠑⠑⠎⠑⠎⠲⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠠⠃⠥⠖⠁⠇⠕⠄⠎⠀⠠⠍⠊⠇⠅⠒⠀⠠⠥⠎⠫⠀⠏⠗⠊⠍⠜⠊⠇⠽⠀⠿⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠍⠕⠵⠵⠜⠑⠇⠇⠁⠂⠀⠃⠥⠖⠁⠇⠕⠄⠎⠀⠍⠊⠇⠅⠀⠁⠙⠙⠎⠀⠁⠀⠗⠊⠡⠂⠀⠀
+⠀⠀⠉⠗⠂⠍⠽⠀⠞⠑⠭⠞⠥⠗⠑⠲⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
+⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠼⠁
+⠀⠀⠀⠀⠠⠡⠑⠑⠎⠑⠀⠠⠍⠁⠅⠬⠀⠠⠏⠗⠕⠉⠑⠎⠎⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
```

Comment:
Would there be enough space for the heading, if it didn't have to be kept with the following text? Doesn't the heading need a blank line before it anyway?

Reply:
Good point. Generating an example that forced this situation was giving me some trouble. It didn't help that I was hand editing the braille some when I noticed that I had left in some print characters that weren't needed. I will update this change in a future commit.

Reply:
I ended up just adding an extra empty line. So now there are 26 lines on the page (but no one will notice).